### PR TITLE
Fixed comman url issue [issue #264]

### PR DESCRIPTION
**Pull request by prajjwalarora**
## What is the change?
Replaced all Post title parameters with Post Id in endpoints and controller, as post title can be common that is creating ambiguity and causing this issue.

## Related issue?
issue: #264 

## How was it tested?
All the controllers after changes tested on the local system performing related operations.

## Checklist:
Before you create this PR, confirm all the requirements listed below by checking the checkboxes `[x]`:

- [x] Have you followed the [Contribution Guidelines](https://github.com/ALPHAVIO/BlogSite/blob/master/CONTRIBUTING.md) while contributing.
- [x] Have you checked there aren't other open [Pull Requests](https://github.com/ALPHAVIO/BlogSite/pulls) for the same update/change?
- [x] Have you made corresponding changes to the documentation?
- [x] Your submission doesn't break any existing feature.
- [x] Have you tested the code before submission?

## Screenshots or Video:
none

### CODE REVIEW

**Comment by RushanKhan1** [comment]
Hey! Thanks for contributing!
There's a typo in your commit message can you please fix that, meanwhile I'll review you pr.

**Comment by RushanKhan1** [comment]
I reviewed the pr and everything works fine, you have squashed the bug, great work!
So please fix the small typo and then I'll add the ``ready-to-merge`` tag.

**Comment by prajjwalarora** [comment]
@RushanKhan1 I've fixed the typo in the commit message.

**Comment by prajjwalarora** [comment]
@Pranav016 I've fixed the common URL issue in the new structured code. please review.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 1]
```diff
@@ -24,8 +27,21 @@ Add any other context or screenshots about the feature request here.
 
 <hr/>
 
+<<<<<<< HEAD
+<<<<<<< HEAD
+=======
+>>>>>>> adc8be918b298942fea3a2ce16d6be1c504062b9
```
Comment (Pranav016) [inline .github/ISSUE_TEMPLATE/feature_request.md:33]: These need to be removed

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 2]
```diff
@@ -24,8 +27,21 @@ Add any other context or screenshots about the feature request here.
...
+<<<<<<< HEAD
+=======
 
 **Note:**
 * If you want to work on an issue, you should check if it has already been assigned to anyone. **If the issue is free** you can comment `/assign` to get the issue assigned to you.
 * If you are raising a new issue and want to work on it then also you should comment `/assign` under the issue to get it auto assigned.
 * Please **refrain from** adding labels to your issue/pull-request on your own. It is the job of the Project Admin and the Mentors to review your issue/pull-request and add labels accordingly.
+>>>>>>> upstream/master
+=======
+>>>>>>> adc8be918b298942fea3a2ce16d6be1c504062b9
```
Comment (Pranav016) [inline .github/ISSUE_TEMPLATE/feature_request.md:47]: All the ones in this document or any other file that were added due to conflicts have to be removed

**Review by Pranav016 — CHANGES_REQUESTED** [review]
Few changes required.
Code looks fine, Add a screen recording that the site works perfectly when we interact with any post. 
Then we'll merge

**Comment by prajjwalarora** [comment]
@Pranav016 I've removed the text that came from conflicts.
here is the link to the video of the working website locally - [link to drive](https://drive.google.com/file/d/1u7WglOTt9aAFsvW-cWtnPap2v8q_UhmF/view?usp=sharing)

**Pull request merged**